app/controllers/api_controller.rb

```diff
@@ -0,0 +1,11 @@
+class ApiController < ApplicationController
+  skip_before_action :verify_authenticity_token
+  
+  private
+  
+  #authenticate_or_request_with_http_basic implements basic HTTP authentication, which ensures HTTP requests are accompanied by a valid username and password.
+  
+  def authenticated?
+    authenticate_or_request_with_http_basic {|username, password_digest| User.where( username: username, password_digest: password_digest ).present?  }
```

Comment:
also,  make sure there is an user with the credentials you are trying to use with your curl request. username:password should be a real email:password combination found in the database. You can create a user from the rails console if you don't have one already.